test/minikube/minikube_fsfreeze_test.go

```diff
@@ -241,3 +249,123 @@ func TestFsFreezeBackupHook(t *testing.T) {
+			"database.sm.noHotCopy.journalPath.enabled":                  "true",
+			"database.sm.noHotCopy.replicas":                             "1",
+			"database.sm.hotCopy.replicas":                               "0",
+			"database.backupHooks.enabled":                               "true",
+			"database.backupHooks.freezeMode":                            "hotsnap",
+		},
+	})
+
+	// Invoke backup hooks on SM pod
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespaceName)
+	smPod := fmt.Sprintf("sm-%s-nuodb-cluster0-demo-0", sourceDatabaseChartName)
+	backupId := "1234abc"
+
+	// Collect backup hooks logs on test failure
+	testlib.AddDiagnosticTeardown(testlib.TEARDOWN_DATABASE, t, func() {
+		testlib.GetAppLog(t, namespaceName, smPod, "_hooks", &corev1.PodLogOptions{Container: "backup-hooks"})
+	})
+
+	// Collect backup hooks logs before it got restarted
+	go testlib.GetAppLog(t, namespaceName, smPod, "_hooks_pre-restart",
+		&corev1.PodLogOptions{Container: "backup-hooks", Follow: true})
+
+	t.Run("verifyPauseArchiving", func(t *testing.T) {
+		// Pause archiving and defer resume invocation
+		testlib.InvokeBackupHook(t, namespaceName, smPod, "pre-backup/"+backupId)
+		defer testlib.InvokeBackupHook(t, namespaceName, smPod, "post-backup/"+backupId)
+
+		// Verify that NuoDB archiving is paused
+		require.Equal(t, 1, testlib.GetStringOccurrenceInLog(t, namespaceName, smPod,
+			fmt.Sprintf("pause archiving (pause id %s)", backupId), &corev1.PodLogOptions{}),
+			"Expected NuoDB to pause archiving")
+	})
+
+	// Check that backup hook sidecar logged expected messages
+	output, err := k8s.RunKubectlAndGetOutputE(t, kubectlOptions, "logs", smPod, "-c", "backup-hooks")
+	require.NoError(t, err, output)
+	require.Contains(t, output, "Pausing archiving for nuodb process")
+	require.Contains(t, output, "Resuming archiving for nuodb process")
+	require.Equal(t, 1, testlib.GetStringOccurrenceInLog(t, namespaceName, smPod,
+		fmt.Sprintf("resume archiving (pause id %s)", backupId), &corev1.PodLogOptions{}),
+		"Expected NuoDB to resume archiving")
+
+	t.Run("verifyHooksTimeout", func(t *testing.T) {
+		// Pause archiving with a smaller timeout to speed up the test but do
+		// not invoke the resume hook
+		backupId := strings.ReplaceAll(t.Name(), "/", "-")
+		timeout := "5"
+		testlib.InvokeBackupHook(t, namespaceName, smPod, "pre-backup/"+backupId, "timeout", timeout)
+
+		// Wait for the automatic backup canceling and metadata files removal
+		testlib.Await(t, func() bool {
+			return testlib.GetStringOccurrenceInLog(t, namespaceName, smPod,
+				fmt.Sprintf("Canceling backup with ID %s. Timeout after %ss", backupId, timeout),
+				&corev1.PodLogOptions{Container: "backup-hooks"}) > 0
+		}, 10*time.Second)
+		k8s.RunKubectl(t, kubectlOptions, "exec", smPod, "-c", "engine", "--",
+			"sh", "-c", `while [ -f "/var/opt/nuodb/archive/backup.txt" ]; do sleep 1; done`)
```

Comment:
Nit: Why not add a timeout to this loop so, in case of failure, the test can exit cleanly instead of getting killed?

Reply:
Will do.